people-and-planet-ai/timeseries-classification/data_utils.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from datetime import datetime, timedelta
+import os
+import time
+from typing import Dict, Iterable
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+
+import trainer
+
+
+# Duration of a time step interval in the timeseries.
+# Training and prediction data must be resampled to this time step delta.
+TIME_STEP_INTERVAL = timedelta(hours=1)
+
+
+def to_unix_time(timestamp: datetime) -> int:
+    return time.mktime(timestamp.timetuple())
+
+
+def with_fixed_time_steps(input_data: Dict[str, np.ndarray]) -> pd.DataFrame:
+    return (
+        pd.DataFrame(input_data)
+        .assign(timestamp=lambda df: df["timestamp"].map(datetime.utcfromtimestamp))
+        .resample(TIME_STEP_INTERVAL, on="timestamp")
+        .mean()
+        .reset_index()
+        .assign(timestamp=lambda df: df["timestamp"].map(to_unix_time))
+        .interpolate()
+    )
+
+
+def read_data(data_file: str) -> pd.DataFrame:
+    mmsi = os.path.splitext(os.path.basename(data_file))[0]
```

Comment:
In the e2e test it says:

`UserWarning: Input dict contained keys ['timestamp', 'mmsi', 'distance_from_shore'] which did not match any model input. They will be ignored by the model.`

Is this expected?

Reply:
It's a harmless warning that we were passing extra inputs to the model. But good catch on it, I made a slight modification to only pass the inputs the model is expecting to get rid of this warning.